Add HPU Activity Type

astachowiczhabana opened the pull request:
Summary: This change is part of Habana Device enablement in PyTorch.
    Allows user to select HPU as profiling activity.

Fixes #ISSUE_NUMBER

astachowiczhabana committed:
Add HPU Activity Type

    Summary: This change is part of Habana Device enablement in PyTorch.
    Allows user to select HPU as profiling activity.

robieta commented:
LGTM. Once the corresponding PRs in https://github.com/pytorch/kineto are approved and we update the pinned version I'll get this going.

astachowiczhabana commented:
Thank you @robieta PR in kineto is: https://github.com/pytorch/kineto/pull/611

astachowiczhabana commented:
Hi @robieta. The corresponding PR for Kineto is now merged (https://github.com/pytorch/kineto/pull/611) can we also merge this one?

robieta commented:
@pytorchbot rebase

robieta commented on torch/csrc/profiler/kineto_shim.cpp:
```diff
@@ -31,6 +31,10 @@ const std::set<libkineto::ActivityType> cudaTypes = {
+
+const std::set<libkineto::ActivityType> hpuTypes = {
+    libkineto::ActivityType::HPU_OP,
+  };
```
FYI Clang format also wants you to fix this indentation.

robieta commented on torch/csrc/profiler/kineto_shim.cpp:
```diff
@@ -296,6 +303,8 @@ c10::DeviceType deviceTypeFromActivity(libkineto::ActivityType activity_type) {
     case libkineto::ActivityType::PYTHON_FUNCTION:
       return c10::DeviceType::CPU;
+    case libkineto::ActivityType::HPU_OP:
+        return c10::DeviceType::HPU;
```
And this one

robieta commented on torch/csrc/profiler/api.h:
```diff
@@ -15,6 +15,7 @@ namespace impl {
 enum class C10_API_ENUM ActivityType {
   CPU = 0,
   CUDA, // CUDA kernels, runtime
+  HPU,  // HPU kernels
```
I think you have to have exactly one space between the comma and comment. I'm not 100% sure; clang format is complaining but is tight lipped about why.

robieta commented:
All the builds are going to fail until https://github.com/pytorch/pytorch/pull/80974 is merged. We will need to handle https://github.com/pytorch/kineto/pull/628 before that can happen.

robieta reviewed (approved):
Approving. Will merge once CI is passing.

astachowiczhabana committed:
Add HPU Activity Type

        Summary: This change is part of Habana Device enablement in PyTorch.
        Allows user to select HPU as profiling activity.

astachowiczhabana committed:
Add HPU Activity Type

            Summary: This change is part of Habana Device enablement in PyTorch.
            Allows user to select HPU as profiling activity.

astachowiczhabana commented:
Hi @robieta. Merge conflicts and lint issues are fixed. CI won't pass until Kineto submodule is updated to this commit: https://github.com/pytorch/kineto/commit/115f530612ce668f88fc966477e5afdc690679de. Can you please help with this?

astachowiczhabana committed:
Add HPU Activity Type

    Summary: This change is part of Habana Device enablement in PyTorch.
    Allows user to select HPU as profiling activity.
